module/users/helper.js

```diff
@@ -128,8 +130,50 @@ module.exports = class UserHelper {
                     createdAt : -1
                 });
 
-                solutionIds = solutionIds.concat(observationSolutions);
-                entityIds = entityIds.concat(observationEntities);
+                let surveysData = 
+                await surveysHelper.surveyDocuments({
+                    createdBy : userId,
+                    status : messageConstants.common.PUBLISHED
+                },[
+                    "solutionId",
+                    "programId",
+                    "name",
+                    "description",
+                    "status",
+                    "endDate",
+                    "createdAt",
+                    "updatedAt"
+                ]);
+
+                let surveyIds = [];
+                let surveySolutions = [];
+                
+                if ( surveysData.length > 0 ) {
+                    
+                    surveysData.forEach(survey=>{
+                        surveyIds.push(survey._id);
+                        survey["isSurvey"] = true;
+                        programIds.push(survey.programId);
```

Comment:
@deepakr123 - We should not push undefined in an array. Add in above query a filter of program exists true

Reply:
resolved